libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -569,7 +570,7 @@ func insertMirroredWSL2Rule(config configuration) bool {
 	if !config.EnableUserlandProxy || config.UserlandProxyPath == "" {
 		return false
 	}
-	if _, err := netlink.LinkByName("loopback0"); err != nil {
+	if _, err := nlwrap.LinkByName("loopback0"); err != nil {
 		if !errors.As(err, &netlink.LinkNotFoundError{}) {
```

Comment:
Definitely not for this PR, but we could even consider that (internal) package to return `errdefs` types I guess (in this case a `errdefs.NotFound`).

Reply:
The `nlwrap` package currently only wraps the minimum number of `vishvananda/netlink` functions ... only some of the underlying functions can return `ErrDumpInterrupted`, and only the few of those we actually use are wrapped. Methods on `nlwrap.Handle` that don't have replacements in `nlwrap` end up as direct calls to the `vishvananda/netlink` method. The idea is that the caller doesn't need to care whether the specific method is wrapped.

If we wanted to modify error types, I think we'd need to wrap all of the Handle methods and their package-level equivalents (or all of the ones we use), and keep it all in-step with the netlink package. That sounds like a bit of a headache!

Reply:
Ah, yeah, that makes sense